### DISCUSSION TITLE
how sorting and searching when querying for many to many relations or in a nested table, search in multiple table is also not work

### DISCUSSION THREAD

**Discussion by nisarg2023** [supabase/supabase#19252] [Questions] [answered]
`
let query = supabase
      .from("report")
      .select(
        "*,users(full_name,avtar,id),sub_categories!inner(name),dishes(name,dish_media(medias(media_url)))"
      )
      .eq("restaurant_id", restaurant_id)
      .eq("dishes.dish_media.is_official", true)
      .eq("dishes.dish_media.status", "accept")
      .limit(1, {
        foreignTable: "dishes.dish_media",
      })
      .order("order_id", {
        foreignTable: "dishes.dish_media",
        ascending: true,
      });
    if (searchKeyword !== "") {
      let searchString = `%${searchKeyword.trim()}%`;
 
      query = query.or(`description.ilike.${searchString}`);
      query = query.or(`name.ilike.${searchString}`, {
        foreignTable: "sub_categories",
      });
    }
This is my supabase query when i execute 
 query = query.or(`description.ilike.${searchString}`);
      query = query.or(`name.ilike.${searchString}`, {
        foreignTable: "sub_categories",
      });
`

 These two queries are in and condition but I want to execute in or condition how can I do it ?
 how sorting and searching when querying for many to many relations or in a nested table 
I am not able to search and sort data so is there any other solution

**Comment by GaryAustin1** [comment]
Seems like a couple of questions.
For many to many table relation make sure both keys are part of the primary key in the table.

The .order() you show only orders the embedded(foreign) table not the parent.
See  "order by related tables" here https://supabase.com/blog/postgrest-11-prerelease

.or() must have all the filters within the same string with each filter triplet separated by a comma.
![image](https://github.com/supabase/supabase/assets/54564956/3395b1b5-4b8f-4d1f-9880-403333a720ac)

See https://supabase.com/docs/reference/javascript/or

Also you cannot use .or() between two different tables.  All the filters in the .or() are in the same table.
Note if you chain the .or()'s like you are doing with one filter each they just get anded together that is not the correct syntax for doing .or().

**Reply by nisarg2023** [reply]
@GaryAustin1 

I have a table schema as follows:

```sql
CREATE TABLE restaurant (
    restaurant_id INT PRIMARY KEY,
    restaurant_name VARCHAR(255) NOT NULL,
    location VARCHAR(255),
    cuisine_type VARCHAR(50),
    rating DECIMAL(3, 2)
);

CREATE TABLE users (
    user_id INT PRIMARY KEY,
    username VARCHAR(255) NOT NULL,
    email VARCHAR(255) UNIQUE,
    registration_date DATE
);

CREATE TABLE restaurant_user (
    restaurant_id INT,
    user_id INT,
    PRIMARY KEY (restaurant_id, user_id),
    FOREIGN KEY (restaurant_id) REFERENCES restaurant(restaurant_id),
    FOREIGN KEY (user_id) REFERENCES users(user_id)
);

I am attempting to order the results based on the username column of the users table using Supabase. Here is the query I tried:


`supabase
  .from('restaurant')
  .select(`*, restaurant_user(*, users(username, email))`)
  .order('users.username', { referenceTable: 'restaurant_user', ascending: false });`


However, this query is not producing the expected results. Could you guide me on how to correctly apply ordering in this scenario? If possible, please provide an example for better understanding.

**Reply by GaryAustin1** [reply, marked as answer]
The order can only go 1 level deep on an embedded table.  The format would be  (users(username),{ascending: false}) according to the link I gave. But users would have to have an fk from restaurant. 

You will need to build a view or handle in a table returning postgres function with an rpc call to it.
